client/src/components/Ticketing/ticketingmanager/showings/ShowingUpdated/EventGeneralForm.tsx

```diff
@@ -0,0 +1,161 @@
+import React, {useState} from 'react';
+import {Field, Formik} from 'formik';
+import {InputControl} from './InputControl';
+import {eventGeneralSchema} from './event.schemas';
+import {FormDeleteButton} from './FormDeleteButton';
+import {FormSubmitButton} from './FormSubmitButton';
+import {useEvent} from './EventProvider';
+import {EventImage} from '../../../../../utils/imageURLValidation';
+
+interface EventGeneralFormProps {
+  onSubmit: (event, actions) => void;
+  onDelete: (event) => void;
+  onLeaveEdit?: () => void;
+}
+export const EventGeneralForm = (props: EventGeneralFormProps) => {
+  const {onSubmit, onDelete, onLeaveEdit} = props;
+  const {eventData} = useEvent();
+  const [disabledURL, setDisabledURL] = useState(
+    eventData?.imageurl === 'Default Event Image',
+  );
+
+  const baseValues = {
+    eventname: eventData ? eventData.eventname : '',
+    eventid: eventData ? eventData.eventid : 0,
+    eventdescription: eventData ? eventData.eventdescription : '',
+    imageurl: eventData ? eventData.imageurl : '',
+    active: eventData ? eventData.active : true,
+    seasonid_fk: eventData ? eventData.seasonid_fk : 7,
+  };
+
+  return (
+    <Formik
+      initialValues={baseValues}
+      onSubmit={onSubmit}
+      validationSchema={eventGeneralSchema}
+    >
+      {({handleSubmit, values, setFieldValue}) => (
```

Comment:
Can you clarify where the these values are coming from? I initially thought it came as props but there is no reference to them. Maybe something with Formik that I don't know, not sure.

Reply:
The Formik component acts as a provider and those are part of the context it creates. Values is the current value for each input, handle submit validates using the validation schema/submits using the on submit prop, and setFieldValue is provided method that allows for manually setting a field value.

Reply:
Oh okay I see, thank you!